### ISSUE TITLE
Training and inference discussion for active learning round 2

### ISSUE THREAD

**Issue by rohanbanerjee** [sct-pipeline/fmri-segmentation#38]
Continuation from the previous round of training: https://github.com/sct-pipeline/fmri-segmentation/issues/35

#### What is the round 2 model
The model which was fine-tuned on the manually corrected segmentations as per the QCs mentioned in #35  is the `round 2` model. A total of 26 images were added in the training of this model since we fine-tuned the previously trained `round 1` model. 

A list of subjects used for the `fine-tuning` is below:  [finetuning.yml](https://github.com/sct-pipeline/fmri-segmentation/files/14950264/qc_fail.yml.zip)

The config (containing preprocessing, hyperparameters) for nnUNetv2 training is:  [plans.json](https://github.com/sct-pipeline/fmri-segmentation/files/14950267/plans.json)

After the training was completed, I ran inference on the rest of the images whose segmentations have to be included in the consequent rounds of training (226 images), below is the QC. 40 subjects from these images will be chosen and included in the `round 3` of training:

[qc_round2_inference.zip](https://github.com/sct-pipeline/fmri-segmentation/files/14950573/qc_round2_inference.zip)

The steps to reproduce the above QC results (/run inference) are the following:
1. Clone this repos
2. `cd fmri-segmentation`
3. Download the model weights (the whole folder) from the link: https://drive.google.com/drive/folders/1WSn-15wGWz6i2_aZeQTwKls2sZ6dpfHf?usp=share_link
4. Install dependencies
```
pip install -r run_nnunet_inference_requirements.txt
```
5. Run the command:
```
python run_nnunet_inference.py --path-dataset <PATH TO FOLDER CONTAINING IMAGES, SUFFIXED WITH _0000> --path-out <PATH TO O/P FOLDER> --path-model <PATH TO DOWNLOADED WEIGHTS FOLDER>
``` 

Next steps:
- [x] Add inference QCs in this comment
- [x] Choose and correct 40 segmentations and include it in the next training round along with the ones that have good segmentation after inference.
- [x]  Get metric by running evaluation on the `held-out test set` (#33)

**Comment by rohanbanerjee** [comment]
The list of 40 subjects chosen for manual corrections is below:
[qc_fail.yml.zip](https://github.com/sct-pipeline/fmri-segmentation/files/14960103/qc_fail.yml.zip)


The QC report for the segmentations manually corrected by me for the above 40 subjects is below:

[qc_round_2_corrected.zip](https://github.com/sct-pipeline/fmri-segmentation/files/14960100/qc_round_2_corrected.zip)


@jcohenadad , I would like your inputs for the above manually corrected images. I will use these images for the next round of training.

CC: @MerveKaptan

**Comment by rohanbanerjee** [comment]
Julien: Redo KCL subjects

**Comment by jcohenadad** [comment]
here you go: [qc_flags.json](https://github.com/sct-pipeline/fmri-segmentation/files/14961529/qc_flags.json)

two issues only: 

![image](https://github.com/sct-pipeline/fmri-segmentation/assets/2482071/04ac64e5-56ce-4a17-9734-8b633e80d2a3)

**Comment by rohanbanerjee** [comment]
Closing the issue since the round 2 training was successfully completed (including running inference and manual correction)
